### DISCUSSION TITLE
How to access Auth (login/ logout/ hash verification and otp verification) services from inside directus endpoints

### DISCUSSION THREAD

**Discussion by mohamedhassen-jahha** [directus/directus#16050] [Archive]
Hey All,

I'm working on a project where i want to make a custom login route with additional security options, in my case One user logged per session at the time. 

I could not find any doc that refers to what i wanted. the doc where about how to use the sdk. (may you please add doc about the  endpoint register function context object properties).

For the moment i managed to achieve the wanted result and everything works perfectly but i had to use `"@directus/sdk"` to get some features Like login and password verification:

`const { Directus } = require("@directus/sdk"); `
`const directus = new Directus(process.env.PUBLIC_URL);`

To get the hash verification i used:
`await directus.utils.hash.verify(password,hash);`

and for the login:
`await directus.auth.login(credentials)`

This works perfectly if i intend to use my route as external API, but i intend to overwrite the /auth/login Route and execute the one I added instead of for both my external app and the directus dash. i believe that could be achieved using hooks where i intercept the request and i redirect the payload to the other route, unfortunately in my case if i do that i will be stuck in a loop because directus.auth.login is basically sending a request to /auth/login with the credentials (My conclusion may be wrong)

So how would i access internal services (the auth ones) and how to overwrite the main auth route?

Thanks.

**Comment by rijkvanzanten** [comment]
You might be able to alter some of the logic through hooks (https://docs.directus.io/extensions/hooks.html#available-events), but the current setup really isn't built for completely replacing the /auth/login endpoint. If you hack around that too much, you'll most likely end up damaging the app usability, as it expects that endpoint to work in a certain way. That being said, I think you can implement your use case by adding `auth.login` filter hook, and cancelling the login if the session already exists for the user. That way, you keep using the regular endpoints, but you add some additional security 👍🏻
